wpilibNewCommands/src/main/native/cpp/frc2/command/CommandPtr.cpp

```diff
@@ -168,6 +168,11 @@ CommandPtr CommandPtr::OnlyIf(std::function<bool()> condition) && {
   return std::move(*this).Unless(std::not_fn(std::move(condition)));
 }
 
+CommandPtr CommandPtr::WithDeadline(CommandPtr&& deadline) && {
+  AssertValid();
+  return std::move(deadline).DeadlineFor(std::move(*this));
```

Comment:
Impl should match Java

Reply:
Do you mean constructing a ParallelDeadlineGroup instead of directly calling deadlineFor?

Reply:
Yes

Reply:
The alternative is changing Java to use `deadlineFor()`.

Reply:
I chose Java over C++ intentionally, though honestly now I don't really remember why.

`beforeStarting`/`andThen` don't delegate one to another, if it means something.